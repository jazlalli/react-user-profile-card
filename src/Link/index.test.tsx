import React from "react";
import { render, screen } from "@testing-library/react";
import { Link } from "./";

test("a link is rendered", () => {
  const { container } = render(<Link href="https://bbc.co.uk">NEWS</Link>);

  expect(screen.getByText("NEWS")).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot(`
    <a href="https://bbc.co.uk">NEWS</a>`);
});

test("a link is rendered with target=_blank and noreferrer noopener", () => {
  const { container } = render(
    <Link href="https://bbc.co.uk" openInNewTab>
      NEWS
    </Link>
  );

  expect(screen.getByText("NEWS")).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot(`
    <a href="https://bbc.co.uk" target="_blank" rel="noreferrer noopener">NEWS</a>`);
});
