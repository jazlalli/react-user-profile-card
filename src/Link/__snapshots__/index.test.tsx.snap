// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a link is rendered with target=_blank and noreferrer noopener: 
    <a href="https://bbc.co.uk" target="_blank" rel="noreferrer noopener">NEWS</a> 1`] = `
<a
  href="https://bbc.co.uk"
  rel="noreferrer noopener"
  target="_blank"
>
  NEWS
</a>
`;

exports[`a link is rendered: 
    <a href="https://bbc.co.uk">NEWS</a> 1`] = `
<a
  href="https://bbc.co.uk"
>
  NEWS
</a>
`;
